### ISSUE TITLE
FRP book, javascript code refactoring

### ISSUE THREAD

**Issue by ca11ado** [SodiumFRP/sodium#169]
If it's okay, I would like to make a javascript code refactoring in the FRP book examples.
It would be great to bring to it ES6 syntax and Airbnb code style. But at least I want to make it more "functional": replace `for` loops with `map` and `reduce`, remove mutable variables and so on.

I need approval, to start the job)

**Comment by the-real-blackh** [comment]
@ca11ado Yes, you're most welcome to do that. Please check your work in to the Git project alongside the original. (So that the original is the same as what's in the book.)
